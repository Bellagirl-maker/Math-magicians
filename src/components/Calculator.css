* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.calculator {
  margin: 5vw 35vw 5vw;
}

.display {
  padding: 1rem;
  background-color: rgba(0, 0, 0, 0.3);
  display: flex;
  justify-content: flex-end;
  color: white;
  font-size: 1.2rem;
}

.operators {
  display: flex;
}

.row {
  display: flex;
}

.Keys {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 1rem;
  border: solid 0.5px #ddd;
  font-size: 1rem;
  width: 5rem;
  flex: 1;
}

.zero {
  flex: 2.4;
}

.Keys:hover {
  cursor: pointer;
}

.Keys:last-child {
  background-color: rgb(250, 158, 29);
}
