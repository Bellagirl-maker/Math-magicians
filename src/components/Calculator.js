import React, { useState } from 'react';
import Keys from './Keys';
import './Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [result, setResult] = useState({
    total: 0,
    next: null,
    operation: null,
  });

  const handleClick = (button) => {
    setResult(calculate(result, button));
  };

  const calcBoard = [
    ['AC', '+/-', '%', 'รท'],
    ['7', '8', '9', 'x'],
    ['4', '5', '6', '-'],
    ['1', '2', '3', '+'],
    ['0', '.', '='],
  ];

  return (
    <div className="calc">
      <h1 className="header">Lets do some Math</h1>
      <div className="cal-Container">
        <div className="display">
          <span>
            {result?.total}
            {' '}
            {result?.operation}
            {' '}
            {result?.next}
          </span>
        </div>

        {calcBoard.map((row) => (
          <div key={`line${row[0]}`} className="row">
            {row.map((button) => (
              <Keys
                Key={button}
                value={button}
                handleBtnClick={() => handleClick(button)}
                kColor={button === '0' ? 'zero' : ''}
              />
            ))}
          </div>
        ))}
      </div>
    </div>
  );
};
export default Calculator;
