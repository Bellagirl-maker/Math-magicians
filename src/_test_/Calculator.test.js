import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import '@testing-library/jest-dom/extend-expect';

describe('Should display calculator component', () => {
  it('should render calculator components', async () => {
    render(<Calculator />);
    const clearbtn = screen.getByText('AC');
    const operate = screen.getByText('=');
    expect(clearbtn).toBeInTheDocument();
    expect(operate).toBeInTheDocument();
  });

  it('should display 2 numbers for calculation', async () => {
    render(<Calculator />);
    const btn1 = screen.getByText('8');
    const btn2 = screen.getByText('4');
    const opBtn = screen.getByText('+');

    fireEvent.click(btn1);
    fireEvent.click(opBtn);
    fireEvent.click(btn2);

    const res = screen.getByText('8 + 4');

    expect(res).toBeInTheDocument();
  });
});

describe('Calculator snapshot', () => {
  test('Calculator snapshot should match dom', () => {
    const tree = renderer.create(<Calculator />).toJSON;
    expect(tree).toMatchSnapshot();
  });
});
