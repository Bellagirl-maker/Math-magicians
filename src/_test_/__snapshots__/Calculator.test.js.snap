// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator snapshot Calculator snapshot should match dom 1`] = `[Function]`;
